fixtures/flight-parcel/src/server.tsx

```diff
@@ -66,8 +66,9 @@ async function render(
 
     // Use client react to render the RSC payload to HTML.
     let [s1, s2] = stream.tee();
-    let data = createFromReadableStream<ReactElement>(s1);
+    let data: Promise<ReactElement>;
     function Content() {
+      data ??= createFromReadableStream<ReactElement>(s1);
```

Comment:
FWIW this took me a long time to debug why preinits weren't working. This must be initialized during render so the current request is set in AsyncLocalStorage. Might be nice if there was some kind of error/warning instead of silently not inserting any scripts.

Reply:
An issue is that it's legit to have an optimistic preload or preinit in module scope. E.g. the first time you load a module on the client. The silent ignoring allows those same modules to keep working with SSR or RSC albeit without the preload.

However, this is also related to another issue that it should be possible to consume values from the RSC stream (like formState) before SSR:ing the payload. It's also useful for other routing concerns.

This is also related to console replaying when doing RSC to RSC since the replays have to be in the scope that needs to replay.

Another case is expanding the React.cache scope to longer than just one RSC render.

We have plans to add *something* that allows expanding the scopes for these AsyncLocalStorage contexts. It's just a little unclear the exact shape atm.